README.md

```diff
@@ -1,2 +1,177 @@
+
+### Example
+
+`styles.css`:
+```css
+.foo {
+  color: red;
+}
+```
+
+`index.js`:
+```js
+import styles from './styles.css' with {type: 'css'};
+document.adoptedStyleSheets.push(styles);
+```
+
+## Usage
+
+### Install
+
+```
+npm i -D @justinfagnani/rollup-plugin-css-modules
+```
+
+### Example Rollup cofig
+
+`rollup.config.js`:
+
+```js
+import {cssModules} from 'rollup-plugin-css-modules';
+
+export default {
+  input: 'index.js',
+  plugins: [cssModules()],
+  output: [{
+    file: 'bundle.js',
+    format: 'es'
+  }]
+};
+```
+
+The `cssModules()` plugin factory takes no options as of now. It may accepts options in the future.
+
+## How it works
+
+`rollup-plugin-css-modules` transforms CSS files that are imported with a `{type: 'css'}` import attribute into JavaScript modules that export a `CSSStyleSheet` object created from the CSS source.
+
+The example files above are transformed to:
+
+`styles.css`:
+```js
+const stylesheet = new CSSStyleSheet();
+stylesheet.replaceSync(`.foo {\n  color: red;\n}\n`);
+export default stylesheet;
+```
+
+`index.js`:
+```js
+import styles from './styles.css';
+document.adoptedStyleSheets.push(styles);
+```
+
+This is a transparent and spec compliant transformation. Like native CSS modules, there are no side-effects from loading the CSS, the CSS is parsed only once, and there's a single `CSSStyleSheet` instance shared among all importers.
+
+After the transformation, Rollup can bundle the transformed CSS module into the JavaScript bundle(s) as usual.
+
+### Potential downsides
+
+While the transform is simple and spec compliant, there are two (small) potential downsides to be aware of:
+
+1. The CSS file can't be dynamic. This is a natural consequence of bundling. With native CSS modules you could import a CSS file from a remote URL or from a file path who's contents could change. With bundling the file is fixed at build time.
+2. The exact same CSS file can't be shared across JS imports and HTML imports. This is also a consequence of bundling. With native CSS modules you can load the CSS file in many different ways: import into JavaScript, with a `<link rel=stylesheet>` HTML tag, or some other CSS loader. With bundling you will have to leave a copy of the CSS as a plain .css file in order to load other ways.
+3. The CSS content is parsed as both JavaScript and CSS. This gives the transform a small performance and memory overhead compared to native CSS modules. On projects that use a similar technique of embedding plain CSS strings into JavaScript and creating `CSSStyleSheet` objects, the overhead was measured to be  small, since parsing strings in JavaScript is very fast.
```

Comment:
comma breaks up the flow unnecessarily
```suggestion
3. The CSS content is parsed as both JavaScript and CSS. This gives the transform a small performance and memory overhead compared to native CSS modules. On projects that use a similar technique of embedding plain CSS strings into JavaScript and creating `CSSStyleSheet` objects, the overhead was measured to be  small since parsing strings in JavaScript is very fast.
```